controllers/db/reports/publication.report.search.controller.ts

```diff
@@ -0,0 +1,1453 @@
+        if(pmidList && pmidList.length > 0)
+        {
+            // Get additional data about articles by PMIDS 
+             articleResults = await models.AnalysisSummaryArticle.findAll({ // review commnet: change this to findAll and send only attributes being displayed on UI with attributes
+              where: { pmid: pmidList },
+              subQuery: false,
+              limit: apiBody.limit,
+              offset: apiBody.offset,
+              order:sort,
+              attributes: [`id`, `pmid`, `pmcid`, `publicationDateDisplay`, `publicationDateStandardized`, `datePublicationAddedToEntrez`, `articleTitle`, `articleTitleRTF`, `publicationTypeCanonical`, `publicationTypeNIH`, `journalTitleVerbose`, `issn`, `journalImpactScore1`, `journalImpactScore2`, `articleYear`, `doi`, `volume`, `issue`, `pages`, `citationCountScopus`, `citationCountNIH`, `percentileNIH`, `relativeCitationRatioNIH`, `readersMendeley`, `trendingPubsScore`],
+              benchmark:true
+            });
+        }
+        let articlesDetails:any =[];
+        if(Object.keys(articleResults).length > 0 )
+        {
+          articlesDetails['rows'] = articleResults; 
+        }
+        else
+        {
+        articlesDetails['rows'] = [];
+        }
+       // articleResults = JSON.stringify(articleResults);
+        searchOutput = {
+          ... articlesDetails,
+          count: authorsResults?.count
+        }
+      } else if(!isAuthorFilter && isArticleFilter) {
+        searchOutput = {
+          ...results,
+          count: results?.count
+        }
+      }else if(isAuthorFilter && isArticleFilter){
+        searchOutput = {
+          ...results,
+          count: typeof(results?.count) === "number" ?  results?.count : results?.count?.length
+        }
+      }
+    } else {
+      searchOutput = await models.AnalysisSummaryArticle.findAndCountAll({
+        where: Sequelize.where(
+          Sequelize.fn(
+            "DATEDIFF",
+            Sequelize.literal("CURRENT_DATE"),
+            Sequelize.fn(
+              "STR_TO_DATE",
+              Sequelize.col("datePublicationAddedToEntrez"),
+              "%Y-%m-%d"
+            )
+          ),
+          {
+            [Op.lte]: 30,
+          },
+        ),
+        limit: apiBody.limit,
+        offset: apiBody.offset,
+        distinct : true,
+        order: sort,
+        // attributes: { exclude: ['AnalysisSummaryAuthorId']},
+        attributes: [`id`,`pmid`, `pmcid`, `publicationDateDisplay`, `publicationDateStandardized`, `datePublicationAddedToEntrez`, `articleTitle`, `articleTitleRTF`, `publicationTypeCanonical`, `publicationTypeNIH`, `journalTitleVerbose`, `issn`, `journalImpactScore1`, `journalImpactScore2`, `articleYear`, `doi`, `volume`, `issue`, `pages`, `citationCountScopus`, `citationCountNIH`, `percentileNIH`, `relativeCitationRatioNIH`, `readersMendeley`, `trendingPubsScore`],
+        col:'pmid',
+        benchmark: true
+      });
+    }
+    return searchOutput;
```

Comment:
wow this function is long. Ideally would be nice if this broken up into parts but most likely to late now

Reply:
Sure. will refactor later. Actually joining the tables on demand.